main.js

```diff
@@ -4,64 +4,64 @@
 define(function (require, exports, module) {
     "use strict";
 
-    var CommandManager      = brackets.getModule("command/CommandManager"),
-        Menus               = brackets.getModule("command/Menus"),
-        AppInit             = brackets.getModule("utils/AppInit"),
-        EditorManager       = brackets.getModule("editor/EditorManager"),
-        ExtensionUtils      = brackets.getModule("utils/ExtensionUtils"),
-        BottomDisplay       = require("BottomDisplayPanal"),
-        MarkErrors          = require("errorDisplay"),
-        parser              = require("parser"),
-        results             = [],
-        showingErrors       = false,
-        lastErrorIndex      = -1,
-        BottomDisplayVar;
-
-    ExtensionUtils.loadStyleSheet(module, "main.less");   
-    
+    var CommandManager        = brackets.getModule("command/CommandManager"),
+        Menus                 = brackets.getModule("command/Menus"),
+        AppInit               = brackets.getModule("utils/AppInit"),
+        EditorManager         = brackets.getModule("editor/EditorManager"),
+        ExtensionUtils        = brackets.getModule("utils/ExtensionUtils"),
+        BottomDisplay         = require("BottomDisplayPanal"),
+        MarkErrors            = require("errorDisplay"),
+        parse                 = require("parser"),
+        showingDescription,
+        BottomDisplayVar,
+        errorCache = {};
+
+    ExtensionUtils.loadStyleSheet(module, "main.less");
+
     function main(){
-        var editor = EditorManager.getFocusedEditor();
-
-        if(editor && editor.document.getLanguage()._name === "HTML"){
-            var text   = editor.document.getText();
-            var result = parser(text);
-            
-            if(result.length > 0){
-                if(lastErrorIndex !== -1 && lastErrorIndex !== (result[3] - 1)){
-                    clearAllErrors();
-                }
-                MarkErrors.showGutter(result[3] - 1);
-                MarkErrors.markErrors(result[3] - 1, result[4] - 1, result[1], result[2]);
-                lastErrorIndex = (result[3] - 1);
-                results.push(result);
-            }else{
-                clearAllErrors();
-            }
-            BottomDisplayVar.update(text);
+        var editor = EditorManager.getActiveEditor();
+        var error;
+        var html;
+
+        if(!editor || editor.document.getLanguage()._name !== "HTML") {
```

Comment:
It's concerning that you're having to use a private member here, `_name`.  @K-LV was doing something like this using a public `id` field, I wonder if you could too?